pkg/chart/provider.go

```diff
@@ -0,0 +1,171 @@
+	//TODO: add caching check here
+	p.logger.Debug(fmt.Sprintf("Getting workspace for Kyma '%s'", compSet.version))
+	ws, err := p.wsFactory.Get(compSet.version)
+	if err != nil {
+		return nil, err
+	}
+
+	return p.renderManifests(compSet, ws, opts)
+}
+
+func (p *Provider) renderManifests(compSet *ComponentSet, ws *workspace.Workspace, opts *Options) ([]*components.Manifest, error) {
+	if err := opts.validate(); err != nil {
+		return nil, errors.Wrap(err, "Invalid provider options defined")
+	}
+
+	//get logger
+	logger, err := p.loggerAdapter()
+	if err != nil {
+		return nil, err
+	}
+
+	//get component list
+	compList, err := p.componentList(ws, compSet.components)
+	if err != nil {
+		return nil, err
+	}
+
+	//get overrides
+	builder, err := p.overrides(compSet.components)
+	if err != nil {
+		return nil, err
+	}
+
+	//get templating instance
+	cfg := &config.Config{
+		WorkersCount:                  opts.WorkersCount,
+		CancelTimeout:                 opts.CancelTimeout,
+		QuitTimeout:                   opts.QuitTimeout,
+		HelmTimeoutSeconds:            60,
+		BackoffInitialIntervalSeconds: 3,
+		BackoffMaxElapsedTimeSeconds:  45,
+		Log:                           logger,
+		HelmMaxRevisionHistory:        10,
+		ComponentList:                 compList,
+		ResourcePath:                  ws.ResourceDir,
+		InstallationResourcePath:      ws.InstallationResourceDir,
+		Profile:                       compSet.profile,
+		Verbose:                       p.debug,
+		KubeconfigSource: config.KubeconfigSource{
+			Path:    "",
+			Content: compSet.kubeconfig,
+		},
+		Version: compSet.version,
+	}
+	templating, err := deployment.NewTemplating(cfg, builder)
+	if err != nil {
+		return nil, err
+	}
+
+	return templating.Render()
+}
+
+//componentList will read the component list from the workspace and align the namespaces with the values retrieved from KEB
+func (p *Provider) componentList(ws *workspace.Workspace, comps []*Component) (*config.ComponentList, error) {
+	compList, err := config.NewComponentList(ws.ComponentFile)
+	if err != nil {
+		return nil, err
+	}
+	compsMap := p.comps(comps)
+	for idx, clComp := range compList.Prerequisites {
+		if comp, ok := compsMap[clComp.Name]; ok {
+			if comp.namespace != "" {
+				p.logger.Debug(
+					fmt.Sprintf("Updating namespace of prerequisite-component '%s' with namespace provided by KEB: '%s' => '%s'",
+						clComp.Name, clComp.Namespace, comp.namespace))
+				compList.Prerequisites[idx].Namespace = comp.namespace
+			}
+		}
+	}
+	for idx, clComp := range compList.Components {
+		if comp, ok := compsMap[clComp.Name]; ok {
+			if comp.namespace != "" {
+				p.logger.Debug(
+					fmt.Sprintf("Updating namespace of component '%s' with namespace provided by KEB: '%s' => '%s'",
+						clComp.Name, clComp.Namespace, comp.namespace))
+				compList.Components[idx].Namespace = comp.namespace
+			}
+		}
+	}
+	return compList, nil
+}
+
+func (p *Provider) comps(comps []*Component) map[string]*Component {
```

Comment:
The name of the function can be more explanatory of what it is doing.

Reply:
Very good point... will change it to `componentMap`